src/htm/algorithms/Connections.hpp

```diff
@@ -115,10 +172,35 @@ struct SegmentData {
  * Segments on this cell.
  *
  */
-struct CellData {
+struct CellData : public Serializable {
   std::vector<Segment> segments;
+
+  //Serialization
+  CerealAdapter;
+  template<class Archive>
+  void save_ar(Archive & ar) const {
+    ar(CEREAL_NVP(segments)
+    );
+  }
+  template<class Archive>
+  void load_ar(Archive & ar) {
+    ar( segments);
+  }
+
+  //operator==
+  bool operator==(const CellData& o) const {
```

Comment:
helper structures in connections `SegmentData`, `SynapseData` now implement `Serializable` and `op==`, which simplifies these operations in Connections.